src/cryptography/x509.py

```diff
@@ -976,6 +982,11 @@ def __ne__(self, other):
         return not self == other
 
 
+class SubjectAlternativeName(GeneralNames):
```

Comment:
You mean like this:

---
class SubjectAlternativeName(object):
    def __init__(self, general_names):
        if not all(isinstance(x, GeneralName) for x in general_names):
            raise TypeError(
                "Every item in the general_names list must be an "
                "object conforming to the GeneralName interface"
            )

        self._general_names = GeneralNames(general_names)

    def __iter__(self):
        return iter(self._general_names)

    def __len__(self):
        return len(self._general_names)

    def get_values_for_type(self, type):
        return [i.value for i in self if isinstance(i, type)]

    def __repr__(self):
        return "<SubjectAlternativeName({0})>".format(self._general_names)

    def __eq__(self, other):
        if not isinstance(other, SubjectAlternativeName):
            return NotImplemented

        return self._general_names == other._general_names

    def __ne__(self, other):
        return not self == other
---

This doesn't look like it gives us any benefit but a lot of duplicate code.
The only difference between the GeneralNames class and the
SubjectAlternateName class is the name of the class, seems like the
textbook example for subclassing to me.


On Tue, May 26, 2015 at 8:08 PM, Alex Gaynor <notifications@github.com>
wrote:

> In src/cryptography/x509.py
> <https://github.com/pyca/cryptography/pull/1956#discussion_r31100117>:
>
> > @@ -976,6 +982,11 @@ def __ne__(self, other):
> >          return not self == other
> >
> >
> > +class SubjectAlternativeName(GeneralNames):
>
> Yup! GeneralNames should be a new class that SubjectAltNames composes
> around and proxies the methods, not subclasses.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1956/files#r31100117>.
>